### PR TITLE
Add a little more explanation for honest-but-curious

### LINKED ISSUE #1227: Explain `honest-but-curious`
On https://docs.opendp.org/en/stable/contributor/development-environment.html:

> `honest-but-curious`: Enable to include constructors that are only private if the constructor arguments are honest.

In this context, it's a little mysterious. Would giving examples help? Perhaps:

> For example, user defined functions require this flag because privacy depends on the stability specified by the user. Similarly, population amplification relies on an honest population size.

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #1227

### CODE REVIEW

File: docs/source/contributing/development-environment.rst [hunk 1]
```diff
@@ -74,8 +74,10 @@ Setting a feature changes how the crate compiles:
           - Enable to include constructors that have not passed the vetting process.
         * - ``honest-but-curious``
           - Enable to include constructors that are only private if the constructor arguments are honest.
+            For example, user defined functions require this flag because privacy depends on the stability specified by the user.
+            Similarly, population amplification relies on an honest population size.
```
Comment (Shoeboxam) [inline docs/source/contributing/development-environment.rst:78]:
```suggestion
            For example, population amplification must assume the user is honest and correct in their assertion that the dataset under observation is a simple sample.
            Similarly, library plugins must assume that the privacy or stability of a given function is governed by the given map.
```

**Review by Shoeboxam — CHANGES_REQUESTED** [review]
I moved pop amplification first, as plugins are not quite as user-facing. Nitted wording of amplification because it mainly depends on the sample being a simple sample.

Thanks!

File: docs/source/contributing/development-environment.rst (hunk 1 above)
Comment (mccalluc) [reply docs/source/contributing/development-environment.rst]: Sounds good. Thanks!

**Review by Shoeboxam — APPROVED** [review]
Great!

**Pull request merged**